tools/Resources.py

```diff
@@ -117,3 +117,163 @@ def get_latest_stat_multiple(target, token, uuids, key):
             print("Return code not 200 for " + str(key) + ": " + str(response.json()))
             return False
 
+    def get_latest_property(target, token, uuid, key):
+        url = "https://" + target + "/suite-api/api/resources/" + uuid + "properties"
+        headers = {
+            'Content-Type': "application/json",
+            'Accept': "application/json",
+            'Authorization': "vRealizeOpsToken " + token
+        }
+        disable_warnings(exceptions.InsecureRequestWarning)
+        try:
+            response = requests.get(url,
+                                    verify=False,
+                                    headers=headers)
+        except Exception as e:
+            print("Problem getting stats Error: " + str(e))
+            return False
+
+        if response.status_code == 200:
+            for propkey in response.json()["property"]:
+                if propkey["name"] is not None and propkey["name"] == key:
+                    return propkey["value"]
+        else:
+            print("Return code not 200 for " + str(key) + ": " + str(response.json()))
+            return False
+
+    # if we expect a number without special characters
+    def get_latest_number_properties_multiple(target, token, uuids, propkey):
+
+        url = "https://" + target + "/suite-api/api/resources/properties/latest/query"
+        headers = {
+            'Content-Type': "application/json",
+            'Accept': "application/json",
+            'Authorization': "vRealizeOpsToken " + token
+        }
+        payload = {
+            "resourceIds": uuids,
+            "propertyKeys": [propkey]
+        }
+        disable_warnings(exceptions.InsecureRequestWarning)
+        try:
+            response = requests.post(url,
+                                     data=json.dumps(payload),
+                                     verify=False,
+                                     headers=headers)
+        except Exception as e:
+            print("Problem getting property Error: " + str(e))
+            return False
```

Comment:
can we put this in an internal function to not duplicate this code? 
Could also be done for the stats collector.

Reply:
I've been thinking about that, too. I would like to handle this issue on a separate assignment. I would like to have only one class as interface to the vrops. That means, get token and get adapter will be included in this class. Then I have all methods in one class and can optimize duplicate code.

Reply:
Good point. Please create a task in jira for keeping this in mind.